### ISSUE TITLE
Some generated files contain zero blocks

### ISSUE THREAD

**Issue by andreconrad** [vincentrussell/json-data-generator#26]
### Environment

- **Operating System:** Ubuntu 20.04.1 LTS
- **Java Version:** openjdk version "1.8.0_275"
- **Maven Version:** Apache Maven 3.6.3

**Used source data (filename: example_schema.json):**

```
[
    '{{repeat(10000)}}',
    {
        "id": "{{uuid()}}",
        "name": "Test",
        "attributes": [
            '{{repeat(2)}}',
            {
                "shirt": "{{random("red","yellow","green")}}",
                "pants": "{{random("blue","black","orange")}}"
            }
        ]

    }
]
```

### Executed commands

```bash
git clone https://github.com/vincentrussell/json-data-generator.git
cd json-data-generator
mvn clean package
java -jar ./target/json-data-generator-1.12-SNAPSHOT-standalone.jar -s example_schema.json -d data.json
```

### Problem description

The generated file in this example (data.json) contain zero blocks starting at line 43475.
This behaviour seems to happen when generated data files have a certain size.

The problem also appears when using the library within a maven project.

Browse file content with: `vim data.json +43475`

```
...
    {
        "id": "22f8183e-26c5-4254-9224-36410bb62b2a",
        "name": "Test",
        "attributes": [

            {
                "shirt": "yellow",
                "pants": "blue"
            },

            {
                "shirt": "yellow",
                "pants": "blue"
            }^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@
        ]

    },

    {
        "id": "3120f06c-f67a-4125-80a6-118ca1074bc9",
        "name": "Test",
        "attributes": [
            ^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@   
...                                                                                                                  
```

Show hexdump with: `xxd -s +803800 data.json | less`

```
...
000c43f8: 5d0a 0a20 2020 207d 2c0a 0a20 2020 207b  ]..    },..    {
000c4408: 0a20 2020 2020 2020 2022 6964 223a 2022  .        "id": "
000c4418: 3232 6638 3138 3365 2d32 3663 352d 3432  22f8183e-26c5-42
000c4428: 3534 2d39 3232 342d 3336 3431 3062 6236  54-9224-36410bb6
000c4438: 3262 3261 222c 0a20 2020 2020 2020 2022  2b2a",.        "
000c4448: 6e61 6d65 223a 2022 5465 7374 222c 0a20  name": "Test",. 
000c4458: 2020 2020 2020 2022 6174 7472 6962 7574         "attribut
000c4468: 6573 223a 205b 0a20 2020 2020 2020 2020  es": [.         
000c4478: 2020 200a 2020 2020 2020 2020 2020 2020     .            
000c4488: 7b0a 2020 2020 2020 2020 2020 2020 2020  {.              
000c4498: 2020 2273 6869 7274 223a 2022 7965 6c6c    "shirt": "yell
000c44a8: 6f77 222c 0a20 2020 2020 2020 2020 2020  ow",.           
000c44b8: 2020 2020 2022 7061 6e74 7322 3a20 2262       "pants": "b
000c44c8: 6c75 6522 0a20 2020 2020 2020 2020 2020  lue".           
000c44d8: 207d 2c0a 0a20 2020 2020 2020 2020 2020   },..           
000c44e8: 207b 0a20 2020 2020 2020 2020 2020 2020   {.             
000c44f8: 2020 2022 7368 6972 7422 3a20 2279 656c     "shirt": "yel
000c4508: 6c6f 7722 2c0a 2020 2020 2020 2020 2020  low",.          
000c4518: 2020 2020 2020 2270 616e 7473 223a 2022        "pants": "
000c4528: 626c 7565 220a 2020 2020 2020 2020 2020  blue".          
000c4538: 2020 7d00 0000 0000 0000 0000 0000 0000    }.............
000c4548: 0000 0000 0000 0000 0000 0000 0000 0000  ................
000c4558: 0000 0000 0000 0000 0000 0000 0000 0000  ................
000c4568: 0000 0000 0000 0000 0000 0000 0000 0a20  ............... 
000c4578: 2020 2020 2020 205d 0a0a 2020 2020 7d2c         ]..    },
000c4588: 0a0a 2020 2020 7b0a 2020 2020 2020 2020  ..    {.        
000c4598: 2269 6422 3a20 2233 3132 3066 3036 632d  "id": "3120f06c-
000c45a8: 6636 3761 2d34 3132 352d 3830 6136 2d31  f67a-4125-80a6-1
000c45b8: 3138 6361 3130 3734 6263 3922 2c0a 2020  18ca1074bc9",.  
000c45c8: 2020 2020 2020 226e 616d 6522 3a20 2254        "name": "T
000c45d8: 6573 7422 2c0a 2020 2020 2020 2020 2261  est",.        "a
000c45e8: 7474 7269 6275 7465 7322 3a20 5b0a 2020  ttributes": [.  
000c45f8: 2020 2020 2020 2020 2020 0000 0000 0000            ......
000c4608: 0000 0000 0000 0000 0000 0000 0000 0000  ................
000c4618: 0000 0000 0000 0000 0000 0000 0000 0000  ................
000c4628: 0000 0000 0000 0000 0000 0000 0000 0000  ................
000c4638: 0000 0000 0000 0000 0000 0000 0000 0000  ................
000c4648: 0000 0000 0000 0000 0000 0000 0000 0000  ................
000c4658: 0000 0000 0000 0000 0000 0000 0000 0000  ................
000c4668: 0000 0000 0000 0000 0000 0000 0000 0000  ................
...
```

The zero blocks in the files lead to unnecessarily large files and also cause problems when parsing, for example with gson.

**Comment by vincentrussell** [comment]
Thank you for the detailed information.   I'll fire up a ubuntu VM when I get a chance to see if I can duplicate this.

**Comment by vincentrussell** [comment]
I was able to duplicate your issue.  I'm going to see if I can figure out what's happening.

**Comment by andreconrad** [comment]
I'm glad that you could duplicate my issue. Thank you for your help.

**Comment by vincentrussell** [comment]
This should be patched in v1.12, which I just released.   Hopefully it will be visible in maven central soon.
